internal/controller/pod_controller_test.go

```diff
@@ -0,0 +1,153 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"context"
+	"testing"
+
+	cloudsqlapi "github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/api/v1alpha1"
+	"github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/testhelpers"
+	"github.com/GoogleCloudPlatform/cloud-sql-proxy-operator/internal/workload"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func TestPodWebhookWithDeploymentOwners(t *testing.T) {
+	_, scheme, err := clientBuilder()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Proxy workload
+	p := testhelpers.BuildAuthProxyWorkload(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "project:region:db")
+	addFinalizers(p)
+	addSelectorWorkload(p, "Deployment", "app", "webapp")
+
+	// Deployment that matches the proxy
+	dMatch := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dMatch.ObjectMeta.Labels = map[string]string{"app": "webapp"}
+
+	// Deployment that does not match the proxy
+	dNoMatch := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dNoMatch.ObjectMeta.Labels = map[string]string{"app": "other"}
+
+	// Deployment that matches the proxy with an owner of an unknown kind
+	dWithOwner := testhelpers.BuildDeployment(types.NamespacedName{
+		Namespace: "default",
+		Name:      "test",
+	}, "webapp")
+	dWithOwner.ObjectMeta.Labels = map[string]string{"app": "webapp"}
+	deploymentOwner := &v1.PartialObjectMetadata{
+		TypeMeta:   v1.TypeMeta{Kind: "BuildToolOperatror", APIVersion: "v1"},
+		ObjectMeta: v1.ObjectMeta{Name: "BuildToolOperator", Namespace: "default"},
+	}
+	err = controllerutil.SetOwnerReference(deploymentOwner, dWithOwner, scheme)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	data := []struct {
+		name       string
+		p          *cloudsqlapi.AuthProxyWorkload
+		d          *appsv1.Deployment
+		wantUpdate bool
+	}{
+		{
+			name:       "Deployment Pod with matching Workload",
+			p:          p,
+			d:          dMatch,
+			wantUpdate: true,
+		},
+		{
+			name:       "Deployment Pod with no match",
+			p:          p,
+			d:          dNoMatch,
+			wantUpdate: false,
+		},
+		{
+			name:       "Deployment Pod with unknown owner",
+			p:          p,
+			d:          dWithOwner,
+			wantUpdate: true,
```

Comment:
Should this be false?

Reply:
No, this should be true. The AuthProxyWorkload should act on the Deployment's pods, even if the deployment is owned by a different operator. We have a user who wants the Cloud SQL Proxy Operator to add proxy containers to pods that are related to a deployment managed by another operator.

Reply:
Is there a test covering the change you made to ignoring owners of an unknown kind in that case?

Reply:
Yes, this is that test.